### ISSUE TITLE
Issue Adding CellCycleScoring MetaData in SeuratV5

### ISSUE THREAD

**Issue by pdimitron** [satijalab/seurat#8059]
I'm not sure if anyone else ran into this issue. I'm updating my script, which I wrote for SeuratV4 to accommodate the changes to Seurat V5. In my script, I want to perform CellCycleScoring prior to SCT so I can regress out the cell-cycle score. 

I read some of the other problems people had running CellCycleScoring and AddModule Score, where it seemed that you had to join layers before scoring. I tried to implement this, as shown below.

>merge
An object of class Seurat
19328 features across 50157 samples within 1 assay
Active assay: RNA (19328 features, 0 variable features)
 8 layers present:

merge.join <- JoinLayers(merge[["RNA"]])
merge.join <- NormalizeData(merge.join)
merge.join <- FindVariableFeatures(merge.join)

> merge.join
Assay (v5) data with 19328 features for 50157 cells
First 10 features:
 Sox17, Mrpl15, Lypla1, Tcea1, Atp6v1h, Rb1cc1, 4732440D04Rik, Alkal1,
Pcmtd1, Gm26901
Layers:
 counts, data

> merge.join.cc <- CellCycleScoring(merge.join, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
The following functions and any applicable methods accept the dots: LayerData
Warning: The following features are not present in the object: Mlf1ip, not searching for symbol synonyms
Warning: The following features are not present in the object: Fam64a, Hn1, not searching for symbol synonyms
Error in `[[<-`:
! Cannot add more or less meta data without feature names
Run `rlang::last_trace()` to see where the error occurred.

I got the scoring to work once, but when I tried to re-run the script top-to-bottom I got the following error.

I'm sure I'm making a basic error, but would appreciate any help.

**Comment by erzakiev** [comment]
Not a response but wanted to thank u 4 the tip here
`merge.join <- JoinLayers(merge[["RNA"]])`

because in their [vignette](https://satijalab.org/seurat/articles/essential_commands.html) they have it simplh wrong
`merge.join <- JoinLayers(merge)`
<img width="910" alt="Screenshot 2023-11-22 at 02 18 26" src="https://github.com/satijalab/seurat/assets/13952639/6cfbf998-2bbb-4cf6-82f1-a299101e987d">

**Comment by pdimitron** [comment]
@erzakiev LOL, no problem. I'm not sure if I'm the paragon of programming you make me out to be. I'm still trying to learn V5 myself. I did find a work around, although it feels very clunky. The error seemed to still be similar to other errors dealing with the AddModuleScore function housed within CellCycleScoring. However, the work arounds in those threads didn't help me. #7191 #7942 #8059

merge.join[["joined"]] <- JoinLayers(merge.join[["RNA"]])
merge.join.cc <- CellCycleScoring(merge.join, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE, assay = "joined")

The difference here is I saved the joined RNA layer as a new layer before running CellCycleScoring on that new layer. I then re-split the layers. I think it would be great if CellCycleScoring could be used directly on a SueratV5 object without having to join and unjoin layers.
